internal/controller/controllers/agent_controller.go

```diff
@@ -440,13 +481,26 @@ func (r *AgentReconciler) updateStatus(ctx context.Context, log logrus.FieldLogg
 			agent.Status.ValidationsInfo = newValidationsInfo
 		}
 
-		if h.Progress != nil && h.Progress.CurrentStage != "" {
-			if isNoneDay2Rebooting, err = r.isDay2NonePlatformHostRebooting(ctx, agent, h); err != nil {
-				log.WithError(err).Errorf("Failed to find if agent %s/%s belongs to none platform cluster and is rebooting", agent.Namespace, agent.Name)
+		// Determine whether or not CSRs should be automatically signed.
+		if (cluster != nil) {
+			isSNOCluster := *cluster.HighAvailabilityMode == string(HighAvailabilityModeNone)
+			isDay2Host := r.isDay2Host(ctx, agent, h)
+			isNonePlatformCluster, err := r.isNonePlatformCluster(ctx, agent, h)
+			if (err != nil) {
+				log.WithError(err).Errorf("Failed to find if agent %s/%s belongs to none platform cluster", agent.Namespace, agent.Name)
+				return ctrl.Result{RequeueAfter: defaultRequeueAfterOnError}, nil
+			}
+			isRebooting, err := r.isRebootingHost(ctx, agent, h)
```

Comment:
The logic that was here before is more correct, because the rebooting logic is only relevant to day2